### PR TITLE
Feature/image utilty

### PR DESCRIPTION
**Pull request by TawKhoury**
**Related issue(s):** #0

## Description

**Notes:**

## Checklist

Usability:
  * [x] Developer
  * [ ] Project manager

Tests
  
  * [x] UI snapshot testing 
  * [x] Unit testing
  * [ ] Integration testing
  * [ ] E2E testing

### CODE REVIEW

File: packages/components/htz-components/src/components/Image/Image.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+**Table of Contents**
+
+* [Image component](#Image-component)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+### Image component
+
```
Comment (TxHawks) [inline packages/components/htz-components/src/components/Image/Image.md]: The documentation is supposed to not only show examples of what image can do, but also explain what can be done and how to use it. See the documentation of `<Grid>` for example

File: packages/apps/haaretz.co.il/components/StandardArticle/StandardArticle.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { ArticleBody, } from '@haaretz/htz-components';
+import { Image, } from '@haaretz/htz-components';
+import getComponent from '../../lib/componentFromTemplate';
```
[the remark this answers is not in the document]
Comment (TawKhoury) [reply packages/apps/haaretz.co.il/components/StandardArticle/StandardArticle.js]: Fixed

File: packages/apps/haaretz.co.il/config/development.js [hunk 3]
```diff
@@ -1,4 +1,6 @@
 module.exports = {
   hostIp: process.env.SSH_CONNECTION.split(' ', 4)[2],
   hostname: process.env.HOSTNAME,
+  imgBaseUrl: 'https://images.haarets.co.il/image',
+  domain: 'https://www.haaretz.co.il',
```
[the remark this answers is not in the document]
Comment (TawKhoury) [reply packages/apps/haaretz.co.il/config/development.js]: Done

File: packages/apps/haaretz.co.il/components/StandardArticle/StandardArticle.js [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { ArticleBody, } from '@haaretz/htz-components';
+import { Image, } from '@haaretz/htz-components';
```
[the remark this answers is not in the document]
Comment (TawKhoury) [reply packages/apps/haaretz.co.il/components/StandardArticle/StandardArticle.js]: Fixed

File: packages/components/htz-components/package.json [hunk 5]
```diff
@@ -62,6 +62,8 @@
     "prop-types": "^15.6.0",
     "react-no-ssr": "^1.1.0",
     "react-test-renderer": "^16.2.0",
+    "react-intersection-observer": "^3.0.2",
+    "react-lazyload": "^2.3.0",
```
[the remark this answers is not in the document]
Comment (TawKhoury) [reply packages/components/htz-components/package.json]: True, fixed

File: packages/components/htz-components/src/components/Image/Image.js [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+
+class Image extends React.Component {
+  constructor(props) {
+    super(props);
+    this.getSources = this.getSources.bind(this);
+    this.getDimensions = this.getDimensions.bind(this);
+    this.getSizes = this.getSizes.bind(this);
+    this.state = {
+      isPicture: null,
+    };
```
[the remark this answers is not in the document]
Comment (TawKhoury) [reply packages/components/htz-components/src/components/Image/Image.js]: True that. Fixed

File: packages/components/htz-components/src/components/Image/Image.js [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+import { attrsPropType, } from '../../propTypes/attrsPropType';
+import { parseComponentProp, } from '@haaretz/htz-css-tools';
+
+/** The shape of an `<Image />` component's transforms options */
+const ImgTransfromOptions = PropTypes.shape({
+  /** The image's aspect ratio to use as base crop, default 'full' */
+  aspect: PropTypes.string,
+  /** An array of strings with misc flags to apply to the image url */
+  flags: PropTypes.arrayOf(PropTypes.string),
+  /** The height, in pixels, of the image the generated url points to. This describes the number of pixels in the file (not rendered size!). */
```
[the remark this answers is not in the document]
Comment (TawKhoury) [reply packages/components/htz-components/src/components/Image/Image.js]: Done

File: packages/components/htz-components/src/components/Image/Image.js [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+  height: PropTypes.string,
+  /** Generate a progressive jpeg. default false */
+  isProgressive: PropTypes.bool,
+  /** The image quality, default 'auto' */
+  quality: PropTypes.string,
+  /** An array of strings with misc transforms to apply to the image url */
+  transforms: PropTypes.arrayOf(PropTypes.string),
+  /** The width, in pixels, of the image the generated url points to. This describes the number of pixels in the file (not rendered size!). */
```
[the remark this answers is not in the document]
Comment (TawKhoury) [reply packages/components/htz-components/src/components/Image/Image.js]: Done

File: packages/components/htz-components/src/components/Image/Image.js [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+  /** The image Background color determined by the user, if no bgc prop passed the default background color will be automatically set from theme color */
+  bgcolor: PropTypes.string,
+  /** Image data object from polopoly */
+  data: PropTypes.shape({
+    /** Image alt from polopoly */
+    alt: PropTypes.string.isRequired,
+    /** Holds the image aspects object */
+    aspects: PropTypes.object,
+    /** Image Component: Image array contains one image name, type and version. */
+    /** Picture Component: When imgArray length > 1 then a picture element will render. Each image name, type and version will build a "source" tag */
```
[the remark this answers is not in the document]
Comment (TawKhoury) [reply packages/components/htz-components/src/components/Image/Image.js]: Done

File: packages/components/htz-components/src/components/Image/Image.js [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+    // Check if transforms matches image aspects pattern
+    let aspectMatchResult = true;
+    if (Array.isArray(transforms)) {
+      aspectMatchResult = transforms.every(
+        (element, index, arr) => (index === 0 ? true : element.aspect === arr[index - 1].aspect)
+      );
+    }
+    // check if custom "imgOptions" prop length is not more than 1 to suit 1 data.imgArray prop passed
+    const imgOptionsIsPictureType = Array.isArray(imgOptions) ? imgOptions.length > 1 : false;
+    // final check to determine if the passed props are for image component or we need to pass them throw to render Picture component
```
[the remark this answers is not in the document]
Comment (TawKhoury) [reply packages/components/htz-components/src/components/Image/Image.js]: Done

File: packages/components/htz-components/src/components/Image/Image.md (hunk 1 above)
Comment (TawKhoury) [reply packages/components/htz-components/src/components/Image/Image.md]: Thanks!

**Pull request merged**